libr/debug/p/native/linux/linux_debug.c

```diff
@@ -272,6 +315,49 @@ RDebugReasonType linux_ptrace_event (RDebug *dbg, int pid, int status) {
 }
 #endif
 
+/*
+ * @brief Search for the parent of the newly created task and
+ * handle the pending SIGTRAP with PTRACE_EVENT_*
+ *
+ * @param tid, TID of the new task
+ * @return RDebugReasonType, Debug reason
+ */
+static RDebugReasonType linux_handle_new_task(RDebug *dbg, int tid) {
+	int ret, status;
+	if (dbg->threads) {
+		RDebugPid *th;
+		RListIter *it;
+		// Search for SIGTRAP with PTRACE_EVENT_* in other threads.
+		r_list_foreach (dbg->threads, it, th) {
+			if (th->pid == tid) {
+				continue;
+			}
+			// Retrieve the signal without consuming it with PTRACE_GETSIGINFO
+			siginfo_t siginfo = { 0 };
+			ret = r_debug_ptrace (dbg, PTRACE_GETSIGINFO, th->pid, 0, (r_ptrace_data_t)(size_t)&siginfo);
+			// Skip if PTRACE_GETSIGINFO fails when the thread is running.
+			if (ret == -1) {
+				continue;
+			}
+			if (siginfo.si_signo == SIGTRAP) {
+				// si_code = (SIGTRAP | PTRACE_EVENT_* << 8)
+				int pt_evt = siginfo.si_code >> 8;
+				// Handle PTRACE_EVENT_* that creates a new task (fork/clone)
+				switch (pt_evt) {
+				case PTRACE_EVENT_CLONE:
```

Comment:
Since you started to use `PTRACE_EVENT_*`, could you please add them in [libr/debug/p/native/linux/linux_ptrace.h](https://github.com/radareorg/radare2/blob/master/libr/debug/p/native/linux/linux_ptrace.h)? So it will still be correct for older Glibc, like the one from Debian Etch (see https://github.com/radareorg/radare2/pull/17459 for more information).

Reply:
They were used before this as well, but sure!

Reply:
Added PTRACE_EVENT_FORK, PTRACE_EVENT_VFORK, PTRACE_EVENT_CLONE, PTRACE_EVENT_EXEC, PTRACE_EVENT_VFORK_DONE, PTRACE_EVENT_EXIT